### PR TITLE
[#3] MINOR: Feature/documentation

### PR DESCRIPTION
**Pull request by Robert-W**
This PR satisfies issue #3.  You can view the [README.md file here](https://github.com/Asymmetrik/elastic-querybuilder/tree/feature/documentation).

Add documentation and examples to README.md.

Other minor updates:
- updated package versions
- fixed spelling issues
- fixed JSDOC comments not having correct types listed
- Added *.log to .gitignore
- Added extra test commands so we can run lint and jest separately

### CODE REVIEW

File: src/BaseBuilder.js [hunk 1]
```diff
@@ -124,6 +95,34 @@ class BaseBuilder {
 	}
 
 	/**
+	* @description Add a field that will be used to generate our aggregations
+	* @param {string} type - Type of aggregation to perform
+  * @param {string|Object} field - Field name to aggregate on
```
Comment (andrewmarcus) [inline src/BaseBuilder.js]: Fix spacing

File: src/BaseBuilder.js [hunk 2]
```diff
@@ -124,6 +95,34 @@ class BaseBuilder {
...
+		saveAggs(BaseBuilder, this._aggs, ...arguments);
+		return this;
+	}
+
+	/**
+	* @description Add a field that will be used to generate filtered aggregations
+	* based on your current boolean filters. Use this for accurate facet counts.
+	* @param {Object} agg - Options for the aggregation
+	* @param {number} agg.size - Maximum number of aggregations to include in the response
+  * @param {string} agg.field - Field name to aggregate on
```
Comment (andrewmarcus) [inline src/BaseBuilder.js]: Fix spacing

File: src/BaseBuilder.js [hunk 3]
```diff
@@ -112,7 +83,7 @@ class BaseBuilder {
 	/**
 	* @description Add must_not boolean queries
 	* @param {string} operation - Type of query to perform
-	* @param {TERMS|TEXT} field - Field to apply the query to
+	* @param {TERMS|TEXT|Object} field - Field to apply the query to
```
Comment (andrewmarcus) [inline src/BaseBuilder.js]: Is TERMS|TEXT actually valid jsdoc syntax?  Are those constant string values?

Comment (Robert-W) [inline src/BaseBuilder.js]: Yes, but this actually reminds me I need to update the typedef in the constants file so this works. You can define enums and then use them as params in jsdoc comments.

Comment (Robert-W) [inline src/BaseBuilder.js]: Ok, this is fixed, you can see the enum and typdef declarations in the constants file.

**Pull request merged**